Add language on video mode preference and DMT video modes (#70)
AMWA-TV/nmos-edid-connection-management: docs/EDID to Receiver Caps Mapping.md

- `urn:x-nmos:cap:format:grain_rate` MUST be calculated with _Field Refresh Rate_
- `urn:x-nmos:cap:format:interlace_mode` MUST be set to `progressive`

> Some of STDs are described in [DMT][DMT] with more precise vertical frequencies than _Field Refresh Rate_. Such STDs SHOULD be described with `urn:x-nmos:cap:format:grain_rate` supporting both frame rate values.

Reviewer: Does DMT define just one additional (more precise) rate? If I've understood that correctly maybe rather than "both frame rate values", maybe "both the STD and DMT frame rate values" or "both the imprecise/standard and precise frame rate values" or something similar?

```suggestion
Some STDs are described in [Display Monitor Timing (DMT)][DMT] with more precise vertical frequencies than _Field Refresh Rate_. Such STDs SHOULD be described with `urn:x-nmos:cap:format:grain_rate` supporting both the STD and DMT frame rate values.
```

Author: Yes, it's a good candidate in the suggestion.

DMT describes the most popular video modes in detail from the waveform perspective, STD rather describes video modes how they're named. A video mode can be named `640x480@85Hz` but its precise vertical frequency per DMT is 85.008 Hz.

I don't think the DMT frame rate is what we expect to see in a Flow so I can't say it's better. `the STD and DMT` sounds good.

Reviewer: If one STD rate always corresponds to the same precise DMT rate, i.e. 85 Hz is _always_ just a short hand convention for 85.008... Hz, and not sometimes 85.007 Hz, then what are the pros and cons of  IS-11 just insisting on using one of them?

Author: Sometimes the difference is bigger: e.g. `640x480@60Hz` has 59.94 Hz in DMT which seems a different video mode. E-EDID A2 section 3.9 says `The standard timing 2 byte codes for most VESA DMT definitions are listed in the latest revision of the DMT document`. That probably means that DMT keeps original information, and STD just refers to it which is a good reason for using DMT.

If nobody knows reasons for not doing it, I'll rewrite this paragraph.

Reviewer 2: In this case is it that the correct grain rate is from the DMT? but a device is allowed to return only the nominal STD grain rate if it is not sensitive to the exact timing.

Author: My understanding is that when we describe a video mode via a Flow, it's up to the Device what are technical parameters to achieve the declared `grain_rate` and how constant it is. EDID works with timings which not always correspond to well-known video modes.

Standard Timings section describe only well-known modes as I can understand from its name, so that the video mode name is enough and every manufacturer should know what the timing for this mode is. Some of the timings are described in DMT.

It depends on the grain rate that baseband-IP gateways use for these well-known modes and timings and I don't know what this grain rate is. I guess, that the video modes described in DMT are common not only for the physical world, but also for the networked world but the corresponding timings are common only for the physical world and can't be exactly followed in the networked world. That's why I would choose the shorthands over the exact timings.

We can also mention that it's implementation-specific and give a vendor a choice between these two grain rates.

Author: E-EDID A2 section 3.8 says `Established Timings I & II indicate Factory Supported Modes of VESA DMT as well as other industry de-facto timings that predate EDID` so we have the same situation with Established Timings.

Author: I added Vertical Frequency Discrepancy section to describe it.

[Example](./Examples.md#standard-timings)

#### Detailed Timing Descriptors (18 Byte Descriptors)

> Defined in [E-EDID][E-EDID] section 3.10, there are 4 possible descriptors which can be provided. The first, _Preferred Timing Mode_ is an obligation and MUST have the highest `urn:x-nmos:cap:meta:preference` value amid the Constraint Sets. Detailed Timing Descriptors describing _Native Video Formats_ MUST have higher `urn:x-nmos:cap:meta:preference` values than video modes not marked as native. Any Detailed Timing Descriptors in [CTA-861][CTA-861] Extension Block (section 7.2.1) MUST follow the mapping.

Reviewer: Obligation... does this mean a requirement of E-EDID? From our mapping point of view could we talk about it being "always present"?

I think the MUST here need to have an "unless overridden by the VIC case described below" or something??

```suggestion
Defined in [E-EDID][E-EDID] section 3.10, there are 4 possible descriptors which can be provided. The first, _Preferred Timing Mode_, is always present and the Constraint Set(s) that represent it MUST have the highest `urn:x-nmos:cap:meta:preference` value among the Constraint Sets. Constraint Sets for Detailed Timing Descriptors describing _Native Video Formats_ MUST have higher `urn:x-nmos:cap:meta:preference` values than video modes not marked as native. Any Detailed Timing Descriptors in [CTA-861][CTA-861] Extension Block (section 7.2.1) MUST follow the mapping.
```

Author: Yes, you're right, `always present` is better.

The language connecting this case with the VIC case looks better than anything that appeared in my head. I also thought about extracting `preference` rules to a separate section.

Author: I extracted `preference` rules for DTDs and VICs to a separate section.

#### Detailed Timing Descriptors (18 Byte Descriptors)

> Defined in [E-EDID][E-EDID] section 3.10, there are 4 possible descriptors which can be provided. Each Detailed Timing Descriptors in Base EDID and [CTA-861][CTA-861] Extension Block (section 7.2.1) MUST follow the mapping.

Author: ```suggestion
Defined in [E-EDID][E-EDID] section 3.10, there are 4 possible descriptors which can be provided. Each Detailed Timing Descriptor in Base EDID and [CTA-861][CTA-861] Extension Block (section 7.2.1) MUST follow the mapping.
```

The _Preferred Vertical Rate_ SHOULD be indicated by using a higher `urn:x-nmos:cap:meta:preference` value in Constraint Set(s) describing this value vs. other _Supported Vertical Rates_.

#### Short Video Descriptors

Short Video Descriptor (SVD) format is defined in [CTA-861][CTA-861] section 7.5.1.

It operates with Video Identification Codes (VICs). Each of them is associated with a union of frame width, height and rate and interlace mode. This mapping is defined in [CTA-861][CTA-861] section 4.1.

### Vertical Frequency Discrepancy

[E-EDID][E-EDID] describes video timings which actual vertical frequency may not be equal to the grain rate of the corresponding video mode.

Established Timings, Standard Timings and CVT 3 Byte Codes operate with video mode descriptions with integer vertical frequencies although these video modes correspond to video timings with fractional vertical frequencies (some of these timings are described in [DMT][DMT]).

> Describing such video modes in Receiver Capabilities is implementation specific. `urn:x-nmos:cap:format:grain_rate` in corresponding Constraint Set(s) MUST describe vertical frequency from either the video mode or the video timings and MAY describe both.

Author: ```suggestion
Describing such video modes in Receiver Capabilities is implementation specific and depend on what network stream the Receiver can handle. `urn:x-nmos:cap:format:grain_rate` in corresponding Constraint Set(s) MUST describe vertical frequency from either the video mode or the video timings and MAY describe both.
```

Some VICs ([CTA-861][CTA-861] section 4.1) are marked as associated with two flavours of the same mode: with a vertical frequency that is an integer multiple of 6 Hz and a vertical frequency adjusted by a factor of 1000/1001. `urn:x-nmos:cap:format:grain_rate` in corresponding Constraint Set(s) MUST describe at least one of these vertical frequencies and MAY describe both.

### Video Mode Preference

> The first Detailed Timing Descriptors in [E-EDID][E-EDID], called _Preferred Timing Mode_, is always present and the Constraint Set(s) that represent it MUST have the highest `urn:x-nmos:cap:meta:preference` value among the Constraint Sets.

Author: ```suggestion
The first Detailed Timing Descriptor in [E-EDID][E-EDID], called _Preferred Timing Mode_, is always present and the Constraint Set(s) that represent it MUST have the highest `urn:x-nmos:cap:meta:preference` value among the Constraint Sets.
```

Some VICs ([CTA-861][CTA-861] section 4.1) are marked as associated with two flavours of the same mode: with a vertical frequency that is an integer multiple of 6 Hz and a vertical frequency adjusted by a factor of 1000/1001. `urn:x-nmos:cap:format:grain_rate` in corresponding Constraint Set(s) MUST describe at least one of these vertical frequencies and MAY describe both.

### Video Mode Preference

Constraint Sets for Detailed Timing Descriptors and Short Video Descriptors describing _Native Video Formats_ MUST have higher `urn:x-nmos:cap:meta:preference` values than Constraint Sets for video modes not marked as native.

The Constraint Set for the first Detailed Timing Descriptor in Base EDID, called _Preferred Timing Mode_, or the Constraint Set for the first Short Video Descriptor in the first CTA-861 Extension if it takes precedence ([CTA-861][CTA-861] section 7.5) MUST have the highest `urn:x-nmos:cap:meta:preference` value among the Constraint Sets.

### Color Subsampling

> If an [E-EDID][E-EDID] doesn't have any [CTA-861][CTA-861] Extensions, color subsampling formats MUST be taken from Base EDID, otherwise from the Extenstions.

Reviewer: ```suggestion
If an [E-EDID][E-EDID] doesn't have any [CTA-861][CTA-861] Extensions, color subsampling formats MUST be taken from Base EDID, otherwise from the Extensions.
```

It operates with Video Identification Codes (VICs), each of them is associated with a union of frame width, height and rate and interlace mode. This mapping is defined in [CTA-861][CTA-861] section 4.1.

> If the first VIC takes precedence over _Preferred Timing Mode_ (as described in [CTA-861][CTA-861] section 7.5), then the related Constraint Set MUST have the highest `urn:x-nmos:cap:meta:preference` value amid the Constraint Sets, taking precedence over the Constraint Set related to the _Preferred Timing Mode_. VICs describing _Native Video Formats_ MUST be described with higher `urn:x-nmos:cap:meta:preference` values than video modes not marked as native.

Reviewer: Tiny suggestion... but mainly the issue I have here is how to express that this overrides the MUST above.

```suggestion
If the first VIC takes precedence over _Preferred Timing Mode_ (as described in [CTA-861][CTA-861] section 7.5), then the related Constraint Set MUST have the highest `urn:x-nmos:cap:meta:preference` value among the Constraint Sets, taking precedence over the Constraint Set related to the _Preferred Timing Mode_. VICs describing _Native Video Formats_ MUST be described with higher `urn:x-nmos:cap:meta:preference` values than video modes not marked as native.
```

> Some of VICs are marked as associated with two flavours of the same mode: with a frame rate that is an integer multiple of 6 Hz and a frame rate adjusted by a factor of 1000/1001. Such VICs MUST be described with `urn:x-nmos:cap:format:grain_rate` supporting both frame rate values.

Reviewer: Tiny suggestion...

```suggestion
Some VICs are marked as associated with two flavours of the same mode: with a frame rate that is an integer multiple of 6 Hz and a frame rate adjusted by a factor of 1000/1001. Such VICs MUST be described with `urn:x-nmos:cap:format:grain_rate` supporting both frame rate values.
```
